ymodem/ymodem.go

```diff
@@ -0,0 +1,412 @@
+// portenta-c33-fwuploader-plugin
+// Copyright (c) 2023 Arduino LLC.  All right reserved.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//go:build !darwin
+
+package ymodem
```

Comment:
Did you implement `ymodem` here?

Why the darwin implementation is different?

Reply:
@facchinm

Reply:
We're using our fork at the moment. https://github.com/arduino/go-xmodem